arm/entrypoint.sh

```diff
@@ -2,60 +2,77 @@
+        then
+          echo "RESOURCE_GROUP_LOCATION is not set." >&2
+          exit 1
+        else
+          az group create --name "$AZURE_RESOURCE_GROUP" --location "$RESOURCE_GROUP_LOCATION"
+        fi
+    fi
 
-# Download parameters file if it is a remote URL
+    URI_REGEX="^(http://|https://)\\w+"
+    GUID=$(uuidgen | cut -d '-' -f 1)
 
-if [[ $AZURE_TEMPLATE_PARAM_FILE =~ $URI_REGEX ]]
-then
-  PARAMETERS=$(curl $AZURE_TEMPLATE_PARAM_FILE)
-  echo "Downloaded parameters from ${AZURE_TEMPLATE_PARAM_FILE}"
-else
-  PARAMETERS_FILE="${GITHUB_WORKSPACE}/${AZURE_TEMPLATE_PARAM_FILE}"
-  if [[ ! -e "$PARAMETERS_FILE" ]]
-  then
-    echo "Parameters file ${PARAMETERS_FILE} does not exists." >&2
-    exit 1
-  else
-    PARAMETERS="@${PARAMETERS_FILE}"
-  fi
-fi
+    # Download parameters file if it is a remote URL
 
-# Generate deployment name if not specified
+    if [[ $AZURE_TEMPLATE_PARAM_FILE =~ $URI_REGEX ]]
+    then
+      PARAMETERS=$(curl $AZURE_TEMPLATE_PARAM_FILE)
+      echo "Downloaded parameters from ${AZURE_TEMPLATE_PARAM_FILE}"
+    else
+      PARAMETERS_FILE="${GITHUB_WORKSPACE}/${AZURE_TEMPLATE_PARAM_FILE}"
+      if [[ ! -e "$PARAMETERS_FILE" ]]
+      then
+        echo "Parameters file ${PARAMETERS_FILE} does not exists." >&2
+        exit 1
+      else
+        PARAMETERS="@${PARAMETERS_FILE}"
+      fi
+    fi
 
-if [[ -z "$DEPLOYMENT_NAME" ]]
-then
-  DEPLOYMENT_NAME="Github-Action-ARM-${GUID}"
-  echo "Generated Deployment Name ${DEPLOYMENT_NAME}"
-fi
+    # Generate deployment name if not specified
 
-# Deploy ARM template
+    if [[ -z "$DEPLOYMENT_NAME" ]]
+    then
+      DEPLOYMENT_NAME="Github-Action-ARM-${GUID}"
+      echo "Generated Deployment Name ${DEPLOYMENT_NAME}"
+    fi
 
-if [[ $AZURE_TEMPLATE_LOCATION =~ $URI_REGEX ]]
-then
-  az group deployment create -g "$AZURE_RESOURCE_GROUP" --name "$DEPLOYMENT_NAME" --template-uri "$AZURE_TEMPLATE_LOCATION" --parameters "$PARAMETERS"
-else
-  TEMPLATE_FILE="${GITHUB_WORKSPACE}/${AZURE_TEMPLATE_LOCATION}"
-  if [[ ! -e "$TEMPLATE_FILE" ]]
+    # Deploy ARM template
+
+    if [[ $AZURE_TEMPLATE_LOCATION =~ $URI_REGEX ]]
+    then
+      az group deployment create -g "$AZURE_RESOURCE_GROUP" --name "$DEPLOYMENT_NAME" --template-uri "$AZURE_TEMPLATE_LOCATION" --parameters "$PARAMETERS"
+    else
+      TEMPLATE_FILE="${GITHUB_WORKSPACE}/${AZURE_TEMPLATE_LOCATION}"
+      if [[ ! -e "$TEMPLATE_FILE" ]]
+      then
+        echo "Template file ${TEMPLATE_FILE} does not exists." >&2
+        exit 1
+      else
+        az group deployment create -g "$AZURE_RESOURCE_GROUP" --name "$DEPLOYMENT_NAME" --template-file "$TEMPLATE_FILE" --parameters "$PARAMETERS"
+      fi
+    fi
+  elif [[ ${AZURE_RG_COMMAND,,} == 'delete' ]]
   then
-    echo "Template file ${TEMPLATE_FILE} does not exists." >&2
-    exit 1
+    echo "Executing commands to Delete resource group."
+    az group delete -n "$AZURE_RESOURCE_GROUP" --no-wait --yes
```

Comment:
do we need --no-wait experience. I believe, we should wait

Reply:
Spec question here. Do we want to wait for long running clean up operation ?

Reply:
waiting for delete operation now.